src/util/http/HttpClient.cpp

```diff
@@ -62,6 +62,14 @@ HttpClientImpl<StreamType>::HttpClientImpl(std::string_view host,
 // ____________________________________________________________________________
 template <typename StreamType>
 HttpClientImpl<StreamType>::~HttpClientImpl() {
+  // If the stream was upgraded to a websocket connection, try to close it.
+  if (ws_) {
```

Comment:
This is something that you probably have to expose also manually (in a threadsafe way, but ask me for details if necessayr).